compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1169,16 +1169,19 @@ object RefChecks {
         target.nonPrivateMember(sym.name)
         .filterWithPredicate:
           member =>
-          val memberIsImplicit = member.info.hasImplicitParams
-          val paramTps =
-            if memberIsImplicit then methTp.stripPoly.firstParamTypes
-            else methTp.firstExplicitParamTypes
-
-          paramTps.isEmpty || memberIsImplicit && !methTp.hasImplicitParams || {
-            val memberParamTps = member.info.stripPoly.firstParamTypes
-            !memberParamTps.isEmpty
-            && memberParamTps.lengthCompare(paramTps) == 0
-            && memberParamTps.lazyZip(paramTps).forall((m, x) => x frozen_<:< m)
+          val memberIsPublic = (member.symbol.flags & AccessFlags).isEmpty && !member.symbol.privateWithin.exists
+          memberIsPublic && {
```

Comment:
This could be `member.symbol.isPublic` but we can also be more precise and check for accessibility (untested):
```suggestion
          val memberIsAccessible = member.symbol.isAccessibleFrom(target)
          memberIsAccessible && {
```

Reply:
I'll take another look during daylight, as my first swing was during interrupted sleep. I think the use case is an arbitrary use site, so that is the reason to require isPublic. I'll reconsider after coffee.